_LowLevelCode/CPP/cpp_communicator/cpp_communicator.cpp

```diff
@@ -81,14 +149,33 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
         return;
     }
     case(labSend): {
+        pCommunicatorHolder->class_ptr->labSend(data_addresses[0], data_tag[0], is_synchroneous, data_buffer, nbytes_to_transfer);
         break;
     }
     case(labReceive): {
+        pCommunicatorHolder->class_ptr->labReceive(data_addresses[0], data_tag[0], is_synchroneous, plhs, nlhs);
         break;
     }
     case(labProbe): {
+        std::vector<int32_t> address_present, tag_present;
+        pCommunicatorHolder->class_ptr->labProbe(data_addresses, data_tag, address_present, tag_present);
+        size_t n_present= address_present.size();
+        if (n_present == 0) {
+            plhs[(int)labProbe_Out::addr_tag_array] = mxCreateNumericMatrix(1, 0, mxINT32_CLASS, mxREAL);
+        }
+        else {
+            plhs[(int)labProbe_Out::addr_tag_array] = mxCreateNumericMatrix(2, n_present, mxINT32_CLASS, mxREAL);
+            int32_t* pAddrTag = (int32_t*)mxGetData(plhs[(int)labProbe_Out::addr_tag_array]);
+            for(size_t i=0;i< n_present;i++){
+                pAddrTag[2 * i + 0] = address_present[i]+1;
+                pAddrTag[2 * i + 1] = tag_present[i];
+            }
+        }
         break;
     }
+    case(clearAll): { // receive and discard all messages, directed to the framework
+        pCommunicatorHolder->class_ptr->clearAll();
```

Comment:
no `break` -- should not rely on fallthrough

Reply:
fixed